### PR TITLE
Mixed precision support

### PR DESCRIPTION
**Pull request by reedwm**
Add mixed precision support to the Keras Resnet model.

This uses the experimental tf.keras mixed precision API. The API will eventually allow users to use mixed precision with a single line of code, but currently it requires several changes to model code.

I tested mixed precision converges with the command line:

```
python keras_imagenet_main.py --clean --num_gpus=8 --data_dir=/raid/imagenet-2012-tfrecord --batch_size=1024 --train_epochs=90 --model_dir ~/fp16_keras_mod --dtype=fp16 --enable_eager=true --datasets_num_private_threads=14
```

The final validation accuracy is 76.35%. I ran tensorflow at commit tensorflow/tensorflow@23ec33ae1418d56c3ce0294720782b636595a2b3.

### CODE REVIEW

File: official/resnet/keras/resnet_model.py [hunk 1]
```diff
@@ -232,10 +232,12 @@ def resnet50(num_classes):
 
   x = layers.GlobalAveragePooling2D(name='avg_pool')(x)
   x = layers.Dense(
-      num_classes, activation='softmax',
+      num_classes,
       kernel_regularizer=regularizers.l2(L2_WEIGHT_DECAY),
       bias_regularizer=regularizers.l2(L2_WEIGHT_DECAY),
       name='fc1000')(x)
+  x = backend.cast(x, 'float32')
```
Comment (azaks2) [inline official/resnet/keras/resnet_model.py]: add a comment and a todo to remove the code when layers have fp16 support
